SDWebImage/Core/SDWebImageDefine.h

```diff
@@ -201,6 +201,11 @@ typedef NS_OPTIONS(NSUInteger, SDWebImageOptions) {
      * Use this flag to transform them anyway.
      */
     SDWebImageTransformVectorImage = 1 << 23,
+    
+    /**
+    * By default, when you use `SDWebImageTransition` to do some view transition after the image load finished, this transition is applied for image download from the network and from disk cache.
+    */
+    SDWebImageTransitionFromDiskCache = 1 << 24,
```

Comment:
This description does not match what `By default` behavior.

By default, when you don't pass any options, transition applys for cache miss (not always network, pay attention to this concept).
When you apply `SDWebImageForceTransition`, applys for any cache hit.
This new option, applys for just memory cache hit.

Reply:
And, can  you change this API st ? I don't want to hack to associated to anything like disk/memory/network. A better way is to provide `SDImageCacheType` this enum to adjust behavior.

For example, can we provide a NSSet or NSArray of `SDImageCacheType` ? We then match anything listed in the set.
Or, can we add something can be used for options bit combination, to control the behavior. This options only handle the case if you want `[disk, network]` to do transition. What about user want `[memory, network]`, `[memory, disk]`, etc ?

Reply:
I guess, what you want is not just about memory/disk, you don't care about this.

You care about `whether this callback is sync or async`, right ? If sync, the image does not cause a flashing because of placeholder replacement.

Reply:
I agree your idea and I polish this API better.